### PR TITLE
design for webhook administration

### PR DESCRIPTION
**Pull request by NotJohs**
PR [#2049](https://github.com/opencollective/opencollective/issues/2049)

### CODE REVIEW

**Comment by Betree** [comment]
@NotJohs why opening a new PR instead of using https://github.com/opencollective/opencollective-frontend/pull/2106? It's useful for us to keep the all the comments in a single place whenever possible.

**Comment by NotJohs** [comment]
@Betree
I deleted that branch. 
There were a lot of things wrong with that branch and also my branch was pushing to the upstream/webhook-administration branch which was out of synch and it created a lot of messed up git commit history. 
I preferred everything to be clean so I started fresh. Pardon me.

**Comment by Betree** [comment]
@NotJohs We can always squash commits if the history is too long :slightly_smiling_face: Amending commits is also a nice feature to avoid that: https://nathanhoad.net/git-amend-your-last-commit/

File: src/components/EditWebhooks.js [hunk 1]
```diff
@@ -189,48 +229,57 @@ class EditWebhooks extends React.Component {
...
-        <div className="editWebhooksActions">
-          <Button bsStyle="primary" onClick={() => this.addWebhook()}>
+        <h2>{this.props.title}</h2>
+        <StyledHr />
+
+        <div className="webhooks">{webhooks.map(this.renderWebhook)}</div>
+
+        {webhooksCount > 0 && <StyledHr />}
+
+        <Box width={[0.9, 1.5 / 2]} mx={['auto', 0, 0]} my={3} className="editWebhooksActions">
```
Comment (Betree) [inline src/components/EditWebhooks.js]: Why `1.5 / 2` and not `0.75` (to match the format of `0.9` next to it)?

Comment (Betree) [inline src/components/EditWebhooks.js]: `mx={['auto', 0, 0]}` can be written with `mx={['auto', 0]}`, the last value is always the one with the biggest priority.

**Comment by NotJohs** [comment]
@Betree thank you. using git-it right now.

File: src/components/EditWebhooks.js (hunk 1 above)
Comment (Betree) [inline src/components/EditWebhooks.js]: Do we need this `className`?

File: src/components/EditWebhooks.js [hunk 2]
```diff
@@ -152,34 +161,65 @@ class EditWebhooks extends React.Component {
...
+              onChange={({ target }) => this.editWebhook(index, url.name, target.value)}
             />
-          ))}
-        </Form>
-      </div>
+          </Box>
+          <Box>
+            <Span fontSize="Paragraph">{activity.label}</Span>
+            <StyledSelect
+              id={activity.name}
```
Comment (Betree) [inline src/components/EditWebhooks.js]: Do we need this `id`?

File: src/components/EditWebhooks.js [hunk 3]
```diff
@@ -68,7 +76,10 @@ class EditWebhooks extends React.Component {
         id: 'webhooks.remove',
         defaultMessage: 'Remove webhook',
       },
-      loading: { id: 'loading', defaultMessage: 'loading' },
+      'webhooks.save': {
+        id: 'webhooks.save',
+        defaultMessage: 'Save {count} webhooks',
+      },
       save: { id: 'save', defaultMessage: 'save' },
```
[the remark this answers is not in the document]
Comment (NotJohs) [reply src/components/EditWebhooks.js]: True.

File: src/components/EditWebhooks.js [hunk 4]
```diff
@@ -152,34 +161,65 @@ class EditWebhooks extends React.Component {
...
-              onChange={value => this.editWebhook(index, field.name, value)}
-              required={field.required}
+      <Flex
+        py={4}
+        key={index}
+        width={[0.9, 1, 1]}
+        mx={['auto', 0, 0]}
+        px={[0, 3, 0]}
+        flexWrap="wrap"
+        className="webhook"
```
[the remark this answers is not in the document]
Comment (NotJohs) [reply src/components/EditWebhooks.js]: No. It was never used. I will remove it.

File: src/components/EditWebhooks.js [hunk 5]
```diff
@@ -152,34 +161,65 @@ class EditWebhooks extends React.Component {
...
+            {intl.formatMessage(this.messages['webhooks.remove'])}
+          </StyledButton>
+        </Box>
+
+        <Box width={[1, 1.5 / 2]}>
+          <Box mb={4}>
+            <Span fontSize="Paragraph" css={'margin-bottom: 1rem'}>
+              {url.label}
+            </Span>
+            <Span fontSize="3rem" css={'color: #D7D9E0; transform: translate(-50px, 23px); position: absolute;'}>
```
[the remark this answers is not in the document]
Comment (NotJohs) [reply src/components/EditWebhooks.js]: Understood.

File: src/components/EditWebhooks.js [hunk 6]
```diff
@@ -152,34 +161,65 @@ class EditWebhooks extends React.Component {
...
+              {index + 1}
+            </Span>
+            <StyledInputGroup
+              key={url.name}
+              type={url.type}
+              name={url.name}
+              label={url.label}
+              prepend="http://"
```
[the remark this answers is not in the document]
Comment (NotJohs) [reply src/components/EditWebhooks.js]: Understood.

File: src/components/EditWebhooks.js [hunk 7]
```diff
@@ -152,34 +161,65 @@ class EditWebhooks extends React.Component {
...
+        <Box width={[1, 1.5 / 2]}>
+          <Box mb={4}>
+            <Span fontSize="Paragraph" css={'margin-bottom: 1rem'}>
+              {url.label}
+            </Span>
+            <Span fontSize="3rem" css={'color: #D7D9E0; transform: translate(-50px, 23px); position: absolute;'}>
+              {index + 1}
+            </Span>
+            <StyledInputGroup
+              key={url.name}
```
[the remark this answers is not in the document]
Comment (NotJohs) [reply src/components/EditWebhooks.js]: An oversight.

File: src/components/EditWebhooks.js (hunk 2 above)
Comment (NotJohs) [reply src/components/EditWebhooks.js]: No.

File: src/components/EditWebhooks.js (hunk 1 above)
Comment (NotJohs) [reply src/components/EditWebhooks.js]: I was counting base on the view scale of 1x 2x 3x. 
Yah, I would change it. It's more readable at 0.75.

Comment (NotJohs) [reply src/components/EditWebhooks.js]: Understood.

**Comment by NotJohs** [comment]
@Betree I having an issue getting the magic link mail

**Comment by Betree** [comment]
@NotJohs If you're working with staging API (default configuration) you should be able to receive them on your email address (check your spam folders).

If using a local API, check your API's console log.

File: components/EditWebhooks.js [hunk 8]
```diff
@@ -152,34 +173,63 @@ class EditWebhooks extends React.Component {
...
-              required={field.required}
+      <Flex
+        py={4}
+        key={index}
+        width={[0.9, 1]}
+        mx={['auto', 0]}
+        px={[0, 3, 0]}
+        flexWrap="wrap"
+        flexDirection="row-reverse"
+        bg={['red', 'blue', 'green']}
```
Comment (Betree) [inline components/EditWebhooks.js]: I don't think this is actually used

```suggestion
```

**Comment by NotJohs** [comment]
@Betree thanks. all done.

**Pull request merged**